src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -128,6 +149,11 @@ public class TopicPartitionChannel {
   private final AtomicLong offsetSafeToCommitToKafka;
 
   // added to buffer before calling insertRows
+  // Reset to earlier value if we were not successfully able to call insertRows.
+  // Do we even need this processedOffset
+  // Why do we even need to rely on a value which is cached in KC?
+  // Source of truth is only Snowflake.
+  // May be remove this and just rely on this boolean (hasChannelReceivedAnyRecordsBefore)
   private final AtomicLong processedOffset; // processed offset
```

Comment:
? Should we just remove this then? This is a confusing set of comments, it's unclear if this is a `TODO` or `I'm not sure what's going on here` type of statement

Reply:
Sorry, I didn't clean this. 🥇 

`processedOffset` is actually needed in case when offsetToken is not updated and it still gives us null but we know what we have processed up until that point. 

will remove the comment/TODO